core/src/test/java/com/sequenceiq/cloudbreak/service/parcel/ParcelFilterServiceTest.java

```diff
@@ -0,0 +1,154 @@
+                        "http://parcel1.com/", OK, "NIFI", Optional.of(getManifestJson("otherService1")), 0 },
+
+                { "Trying to download the parcel manifest in case of BASE image, when we can reach the file which is a standard " +
+                        "manifest file and it DOES contains the component then assign the parcel. (NIFI usecase)",
+                        "http://parcel1.com/", OK, "NIFI", Optional.of(getManifestJson("NIFI")), 1 },
+
+                { "Trying to download the parcel manifest in case of BASE image, when we can reach the file which is a standard " +
+                        "manifest file then it should assign the parcel, because we dont know that it is required, or not. (PROFILER usecase)",
+                        "http://parcel1.com/", OK, "NIFI", Optional.of("{..ewwer"), 1 },
+
+                { "Trying to download the parcel manifest in case of BASE image, when we can NOT reach the manifest file " +
+                        "then it should assign the parcel, because we can't check that it is required, or not. (authentication required usecase)",
+                        "http://parcel1.com/", NOT_FOUND, "NIFI", Optional.empty(), 1 },
+
+                { "Trying to download the parcel manifest in case of PREWARMED image, when we can reach the file which is a standard " +
+                        "manifest file and it DOES NOT contains the component then DO NOT assign the parcel.",
+                        "http://parcel1.com/", OK, "NIFI", Optional.of(getManifestJson("otherService1")), 0 },
+
+                { "Trying to download the parcel manifest in case of PREWARMED image when we can reach the file which is a standard " +
+                        "manifest file and it DOES contains the component then assign the parcel. (NIFI usecase)",
+                        "http://parcel1.com/", OK, "NIFI", Optional.of(getManifestJson("NIFI")), 1 },
+
+                { "Trying to download the parcel manifest in case of PREWARMED image, when we can reach the file which is a standard " +
+                        "manifest file then it should assign the parcel, because we dont know that it is required, or not. (PROFILER usecase)",
+                        "http://parcel1.com/", OK, "NIFI", Optional.of("{..ewwer"), 1 },
+
+                { "Trying to download the parcel manifest in case of PREWARMED image, when we can NOT reach the manifest file " +
+                        "then it should assign the parcel because probably that is not available anymore but the PREWARMED image contains it. " +
+                        "(RELENG deleted a released artifact usecase)",
+                        "http://parcel1.com/", NOT_FOUND, "NIFI", Optional.empty(), 1 },
+        });
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("data")
+    public void test(String description, String parcelUrl, Response.Status status, String serviceNameInTheBlueprint, Optional<String> manifest,
```

Comment:
Nit: since junit5 is used, it is not necessary to use public, can be package private.